### ISSUE TITLE
Chunk grouping

### ISSUE THREAD

**Issue by jakearchibald** [rollup/rollup#5344]
### Feature Use Case

**index.js**

```js
import { specialLogger } from './common-dependency.js';

const lazyThing1Promise = import('./lazy-thing-1.js');
const lazyThing2Promise = import('./lazy-thing-2.js');

specialLogger(lazyThing1Promise, lazyThing2Promise);
```

**lazy-thing-1.js**

```js
import { specialLogger } from './common-dependency.js';
import { somethingElse } from './something-else.js';

specialLogger('hi');

export default 'yo';
```

**lazy-thing-2.js** is much the same.

When deploying to the web, I want the main bundle to contain everything for the first interaction, which includes `specialLogger`. It mustn't include either of the lazy scripts. It mustn't include any dependency _only_ needed by the lazy scripts.

To optimise the loading, I decide that the two lazy scripts should be in a single chunk. It doesn't seem like I can configure Rollup to do this.

[`output.manualChunks`](https://rollupjs.org/configuration-options/#output-manualchunks) is a similar feature, but doesn't fit, because:

> each property represents a chunk that contains the listed modules **and all their dependencies**

That means `common-dependency.js` gets dragged into the lazy chunk, and statically imported, spoiling the whole lazy loading thing.

My current solution is to create an intermediate file:

**index.js**

```js
import { specialLogger } from './common-dependency.js';

const lazyThing1Promise = import('./lazy-thing-1.js').then(m => m.lazyThing1);
const lazyThing2Promise = import('./lazy-thing-2.js').then(m => m.lazyThing2);

specialLogger(lazyThing1Promise, lazyThing2Promise);
```

**lazy.js**

```js
export { default as lazyThing1 } from './lazy-thing-1.js';
export { default as lazyThing2 } from './lazy-thing-2.js';
```

But it feels like I should be able to configure this as part of the build

### Feature Proposal

An output option, similar to `manualChunks`, but doesn't force all dependencies to be part of the chunk.

Or, perhaps `manualChunks` could have a return option:

```js
manualChunks: [
  { id: 'foo', shallow: true }
]
```

Where `shallow: true` means dependencies of `'foo'` will not be forced into the same chunk. They'll only be in the chunk if they're the only used within the chunk.

**Comment by lukastaegert** [comment]
Yes, extending the manualChunks syntax is definitely useful. I have some mid-term plans to offer a solid solution for the "optimize for the initial load" question, but there are many other important things I am working on as well, so this will take time.
If someone wants to try a PR, this would be welcome, but I fear the manual chunk logic is baked into the chunking logic, which is non-trivial, so your luck may vary.

**Comment by tmsns** [comment]
I've been meddling with `manualChunks` a lot the last few months. Would it help your case if you define in `manualChunks` that `specialLogger` should be assigned to the same chunk as `index.js`, next to the assignment of the two lazy chunks in one?

Something in the lines of:
```js
const config = {
  output: {
    manualChunks: function(id) {
      if(id.endsWith('/index.js') || id.endsWith('/common-dependency.js') {
        return 'index.js';
      }
      else if(id.endsWith('/lazy-thing-1.js') || id.endsWith('/lazy-thing-2.js') {
        return 'lazy-thing.js';
      }
      return null;
    },
  },
};
```

**Comment by jakearchibald** [comment]
Yes, but that doesn't scale beyond this simple example. You'd need to know the whole module graph up front.
